sakai-evaluation-impl/src/java/org/sakaiproject/evaluation/logic/externals/EvalExternalLogicImpl.java

```diff
@@ -686,35 +772,97 @@ public int countEvalGroupsForUser(String userId, String permission) {
      */
-    public int countUserIdsForEvalGroup(String evalGroupId, String permission) {
-        // get the count from the method which retrieves all the groups,
-        // this method might be better to retire
-        return getUserIdsForEvalGroup(evalGroupId, permission).size();
+    public int countUserIdsForEvalGroup( String evalGroupID, String permission, Boolean sectionAware )
+    {
+        return getUserIdsForEvalGroup( evalGroupID, permission, sectionAware ).size();
     }
 
     /* (non-Javadoc)
-     * @see org.sakaiproject.evaluation.logic.externals.ExternalEvalGroups#getUserIdsForEvalGroup(java.lang.String, java.lang.String)
+     * @see org.sakaiproject.evaluation.logic.externals.ExternalEvalGroups#getUserIdsForEvalGroup(java.lang.String, java.lang.String, java.lang.Boolean)
      */
-    public Set<String> getUserIdsForEvalGroup(String evalGroupId, String permission) {
-        Set<String> userIds = new HashSet<String>();
-
-        // only go on to check the Sakai sites/groups if nothing was found
-        if (userIds.size() == 0) {
-            String reference = evalGroupId;
-            List<String> azGroups = new ArrayList<String>();
-            azGroups.add(reference);
-            userIds.addAll( authzGroupService.getUsersIsAllowed(permission, azGroups) );
-            // need to remove the admin user or else they show up in unwanted places
-            if (userIds.contains(ADMIN_USER_ID)) {
-                userIds.remove(ADMIN_USER_ID);
+    public Set<String> getUserIdsForEvalGroup( String evalGroupID, String permission, Boolean sectionAware )
+    {
+        // Determine the site ID (strip out the section prefix and ID if present)
+        String siteID = evalGroupID;
+        boolean isSectionGroupID = false;
+        if( siteID.contains( EvalConstants.GROUP_ID_SECTION_PREFIX ) )
+        {
+            isSectionGroupID = true;
+            siteID = siteID.substring( 0, siteID.indexOf( EvalConstants.GROUP_ID_SECTION_PREFIX ) );
+        }
+
+        // If it's not section aware...
+        Set<String> userIDs = new HashSet<>();
+        if( !sectionAware )
+        {
+            // Get the list normally
+            List<String> azGroups = new ArrayList<>();
+            azGroups.add( siteID );
+            userIDs.addAll( authzGroupService.getUsersIsAllowed( permission, azGroups ) );
+            if( userIDs.contains( ADMIN_USER_ID ) )
+            {
+                userIDs.remove( ADMIN_USER_ID );
             }
         }
 
-        return userIds;
-    }
+        // Otherwise, it's section aware but we only need to run the following if the sectin prefix is present in the evalGroupID
+        else if( isSectionGroupID )
+        {
+            // Determine the section ID
+            String sectionID = evalGroupID.substring( evalGroupID.indexOf( EvalConstants.GROUP_ID_SECTION_PREFIX ) + 9 );
+            if( siteID.contains( EvalConstants.GROUP_ID_SITE_PREFIX ) )
```

Comment:
Maybe parse out the siteid and sectionid right off the bat with empty string defaults that you can check against later rather than parsing the string again.